packages/appkit/src/client.ts

```diff
@@ -705,10 +705,9 @@ export class AppKit {
 
     const extendedNetwork = this.extendCaipNetwork(network, this.options)
 
-    ChainController.addNetwork(extendedNetwork)
-
     if (this.caipNetworks && !this.caipNetworks?.find(n => n.id === extendedNetwork.id)) {
       this.caipNetworks.push(extendedNetwork)
+      ChainController.addNetwork(extendedNetwork)
     }
```

Comment:
Maybe nit but:
should we make `this.caipNetworks` directly fetched from chain controller? why keep two separate fields ? 

IMO we should initialize from chaincontroller and read from it. Maybe a getter

Reply:
Agree with the approach, I guess itl'll require further refactor, let's keep them separated